src/DataCollectors/TraceDataCollector/IDataCollection.cs

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TraceCollector
+{
+    using System;
+    using System.ComponentModel;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.DataCollection;
+
+    #region Data Collection wrapping
+
+    /* This wrapping layer is here so that the collector is testable. The classes used by the actual implementation are
+     not publicly constructable and do not use interfaces.*/
+
+#pragma warning disable SA1649 // File name must match first type name
+
+#pragma warning disable SA1402 // File may only contain a single class
+    internal interface IDataCollectionEvents
```

Comment:
Should we make is IDataCollectionEventsWrapper, since we get DataCollectionEvents from TP, which makes is look like that DataCollectionEvents is derived from IDataCollectionEvents, where as actually "DataCollectionEventsWrapper" is derived from IDataCollectionEvents

Reply:
Not doing this for better readability while using this Interfaces.